### PR TITLE
disable commissioner related CoAP resource before accepted

### PR DESCRIPTION
**Pull request by bukepo**
Currently OpenThread implemented functionality as a on-mesh commissioner. As on-mesh commissioner it's also a border agent of itself. However, this built-in border agent functionality always handles requests to `/c/rx` even not started or not accepted as a commissioner. This behavior prevents a thread node being a simple border agent, and also interferes implementing border agent on host side.

This PR adjusted installing commissioner related CoAP resource only after the node becomes active commissioner.

### CODE REVIEW

File: src/core/meshcop/commissioner.cpp [hunk 1]
```diff
@@ -79,14 +79,25 @@ Commissioner::Commissioner(ThreadNetif &aThreadNetif):
...
 {
     memset(mJoiners, 0, sizeof(mJoiners));
+}
+
+otInstance *Commissioner::GetInstance(void)
+{
+    return mNetif.GetInstance();
+}
+
+void Commissioner::SetUpCoap()
```
Comment (jwhui) [inline src/core/meshcop/commissioner.cpp]: nit: rename to `AddCoapResources()` to better describe what this is doing?

File: src/core/meshcop/commissioner.cpp [hunk 2]
```diff
@@ -79,14 +79,25 @@ Commissioner::Commissioner(ThreadNetif &aThreadNetif):
...
+{
     mNetif.GetCoapServer().AddResource(mRelayReceive);
     mNetif.GetCoapServer().AddResource(mDatasetChanged);
     mNetif.GetSecureCoapServer().AddResource(mJoinerFinalize);
 }
 
-otInstance *Commissioner::GetInstance(void)
+void Commissioner::TearDownCoap()
```
Comment (jwhui) [inline src/core/meshcop/commissioner.cpp]: nit: rename to `RemoveCoapResources()` to better describe what this is doing?

**Comment by bukepo** [comment]
Thanks and modified.

File: src/core/meshcop/commissioner.hpp [hunk 3]
```diff
@@ -209,6 +209,9 @@ class Commissioner
         kRemoveJoinerDelay    = 20,     ///< Delay to remove successfully joined joiner
     };
 
+    void AddCoapResources();
+    void RemoveCoapResources();
```
Comment (jwhui) [inline src/core/meshcop/commissioner.hpp]: nit: All void functions or methods shall explicitly declare and specify the void type keyword.

File: src/core/meshcop/commissioner.cpp [hunk 4]
```diff
@@ -79,14 +79,25 @@ Commissioner::Commissioner(ThreadNetif &aThreadNetif):
...
+
+void Commissioner::AddCoapResources()
+{
     mNetif.GetCoapServer().AddResource(mRelayReceive);
     mNetif.GetCoapServer().AddResource(mDatasetChanged);
     mNetif.GetSecureCoapServer().AddResource(mJoinerFinalize);
 }
 
-otInstance *Commissioner::GetInstance(void)
+void Commissioner::RemoveCoapResources()
```
Comment (jwhui) [inline src/core/meshcop/commissioner.cpp]: nit: All void functions or methods shall explicitly declare and specify the void type keyword.

File: src/core/meshcop/commissioner.cpp [hunk 5]
```diff
@@ -79,14 +79,25 @@ Commissioner::Commissioner(ThreadNetif &aThreadNetif):
...
 {
     memset(mJoiners, 0, sizeof(mJoiners));
+}
+
+otInstance *Commissioner::GetInstance(void)
+{
+    return mNetif.GetInstance();
+}
+
+void Commissioner::AddCoapResources()
```
Comment (jwhui) [inline src/core/meshcop/commissioner.cpp]: nit: All void functions or methods shall explicitly declare and specify the void type keyword.

**Pull request merged**